Xamarin.Forms.Platform.WinRT/FormsTextBox.cs

```diff
@@ -237,19 +239,65 @@ static void OnIsPasswordChanged(DependencyObject dependencyObject, DependencyPro
 
 		void OnSelectionChanged(object sender, RoutedEventArgs routedEventArgs)
 		{
-			if (!IsPassword)
-			{
-				return;
-			}
+			// Cache this value for later use as explained in OnKeyDown below
+			_cachedSelectionLength = SelectionLength;
+		}
 
-			// Prevent the user from selecting any text in the password box by forcing all selection
-			// to zero-length at the end of the Text
-			// This simulates the "do not allow clipboard copy" behavior the PasswordBox control has
-			if (SelectionLength > 0 || SelectionStart < Text.Length)
+		// Because the implementation of a password entry is based around inheriting from TextBox (via FormsTextBox), there
+		// are some inaccuracies in the behavior. OnKeyDown is what needs to be used for a workaround in this case because 
+		// there's no easy way to disable specific keyboard shortcuts in a TextBox, so key presses are being intercepted and 
+		// handled accordingly.
+		protected override void OnKeyDown(KeyRoutedEventArgs e)
+		{
+			if (IsPassword)
 			{
-				SelectionLength = 0;
-				SelectionStart = Text.Length;
+				// The ctrlDown flag is used to track if the Ctrl key is pressed; if it's actively being used and the most recent
+				// key to trigger OnKeyDown, then treat it as handled.
+				var ctrlDown = Window.Current.CoreWindow.GetKeyState(VirtualKey.Control).HasFlag(CoreVirtualKeyStates.Down);
+
+				if (ctrlDown && e.Key == VirtualKey.Control)
```

Comment:
This `if` block seems unnecessary. You're checking whether the Ctrl key is down and the Ctrl key is the one being pressed.